librz/core/cfile.c

```diff
@@ -892,32 +892,58 @@ typedef struct {
 	RzBin *bin;
 } RzCoreLinkData;
 
-static bool linkcb(void *user, void *data, ut32 id) {
-	RzCoreLinkData *ld = user;
-	RzIODesc *desc = (RzIODesc *)data;
-
+static bool resolve_import_cb(RzCoreLinkData *ld, RzIODesc *desc, ut32 id) {
 	RzBinFile *bf = rz_bin_file_find_by_fd(ld->bin, desc->fd);
-	if (bf) {
-		RzListIter *iter;
-		RzBinSymbol *sym;
-		RzList *symbols = rz_bin_file_get_symbols(bf);
-		rz_list_foreach (symbols, iter, sym) {
-			if (!strcmp(sym->name, ld->name)) {
-				ld->addr = sym->vaddr;
-				return false;
-			}
+	if (!bf) {
+		return true;
+	}
+	RzListIter *iter;
+	RzBinSymbol *sym;
+	RzList *symbols = rz_bin_file_get_symbols(bf);
+	rz_list_foreach (symbols, iter, sym) {
+		if (!strcmp(sym->name, ld->name)) {
+			ld->addr = sym->vaddr;
+			return false;
 		}
 	}
 	return true;
 }
 
+static bool map_multi_dex(RzCore *core, RzIODesc *desc, ut32 id) {
+	if (!rz_str_endswith(desc->name, ".dex")) {
+		return true;
+	}
+
+	ut64 size = rz_io_desc_size(desc);
+	ut64 baddr = rz_io_map_next_available(core->io, 0x200000, size, 0x100000);
```

Comment:
What are these hardcoded values?

Reply:
is the base address and the advance value
so if it fails `base + advance` till it finds the next empty address space